lib/coinbase/webhook.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a Webhook.
+  class Webhook
+    class << self
+      def create(network_id:, notification_uri:, event_type:, event_filters:)
+        model = Coinbase.call_api do
+          webhooks_api.create_webhook(
+            create_webhook_request: {
+              network_id: Coinbase.normalize_network(network_id),
+              notification_uri: notification_uri,
+              event_type: event_type,
+              event_filters: event_filters
+            }
+          )
+        end
+
+        new(model)
+      end
+
+      def list
+        Coinbase::Pagination.enumerate(lambda(&method(:fetch_webhooks_page))) do |webhook|
+          Coinbase::Webhook.new(webhook)
+        end
+      end
+
+      private
+
+      def webhooks_api
+        Coinbase::Client::WebhooksApi.new(Coinbase.configuration.api_client)
+      end
+
+      def fetch_webhooks_page(page)
+        webhooks_api.list_webhooks({ limit: DEFAULT_PAGE_LIMIT, page: page })
+      end
+    end
+
+    # Returns a new Webhook object.
+    # @param model [Coinbase::Client::Webhook] The underlying Webhook object
+    def initialize(model)
+      raise ArgumentError, 'model must be a Webhook' unless model.is_a?(Coinbase::Client::Webhook)
+
+      @model = model
+    end
+
+    def id
+      @model.id
+    end
+
+    def network_id
+      @model.network_id
+    end
+
+    def notification_uri
+      @model.notification_uri
+    end
+
+    def event_type
+      @model.event_type
+    end
+
+    def event_filters
+      @model.event_filters
+    end
+
+    def update(notification_uri:)
```

Comment:
getting:
```
Coinbase::MalformedRequestError: request body has an error: doesn't match schema: Error at "/event_filters/0": value must be an object
```
might need to do this conversion?
```
      serialized_event_filters = event_filters.map do |filter|
        filter_hash = {}
        filter_hash[:contract_address] = filter.contract_address if filter.contract_address
        filter_hash[:from_address] = filter.from_address if filter.from_address
        filter_hash[:to_address] = filter.to_address if filter.to_address
        filter_hash
      end
```
then we can take:
```
          update_webhook_request: {
            network_id: network_id,
            notification_uri: notification_uri,
            event_type: event_type,
            event_filters: serialized_event_filters
          }
```
i think we can also do this at callee